### PR TITLE
Add the droid onboarding screen

### LINKED ISSUE #7000: Create OnBoarding screen for Android
Do it according to design and bind it with the new view model.

Be aware of errors and show them when appropriate.

#6998 should be done first.

Designs and full specs: https://www.notion.so/toggl/Designs-wip-d767ebc157c542c790e1bc93eeaf006f#835299ea40344044b2572f356b4f47c1

Android designs are pending, but we can start implementation taking iOS designs as reference and then do the appropriate changes later.

**NOTE:** This issue is part of the `Onboarding refactor` project and it should be merged into the project branch and not into develop.

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
When done, this will be the droid onboarding screen implementation.

### Relationships
Awaiting on #7106 
Will close #7000 

## Why do we want this?
To improve the user experience.

## How is it done?
~It's not done yet, but the idea is to have a pager in the background that the user can scroll the onboarding pages and the buttons to login/signup fixed in the screen.~
It's just a surface view that plays the togglman animation every 4 seconds.
Everything else is just a bunch of simple bindings.

### Why not another way?
Feel free to suggest something different.

### Side effects
None.

## Review hints
Feel free to start reviewing by commit.
What's done:
- [x] base layout
- [x] activity bindings
- ~onboarding pager logic~
- ~each onboarding page~
- [x] toggl man play loop

## Localizations
This PR when done, will add new copies. I'll be opening a PR for them when they come.

## :squid: Permissions
🦑 it ~after the draft is upgraded to a PR~

Please make sure that the actual `SignUpViewController.cs` is still in the project and works fine.

### CODE REVIEW

**Comment by pdrj** [comment]
A couple of questions to @daividssilverio. 

When is this screen supposed to appear? On a fresh install it doesn't show. It only triggers after logging out of the app.

Also, when trying to login with google, I get the following crash:
```
2020-02-28 12:09:43.367 25888-25888/com.toggl.giskard.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.toggl.giskard.debug, PID: 25888
    android.runtime.JavaProxyThrowable: System.InvalidOperationException: Failed to find a presenter that could present ViewModel with type TermsAndCountryViewModel
      at Toggl.Core.UI.Navigation.CompositePresenter.Present[TInput,TOutput] (Toggl.Core.UI.ViewModels.ViewModel`2[TInput,TOutput] viewModel, Toggl.Core.UI.Views.IView sourceView) [0x00048] in <d2bbbfc73143447c8d336e7fe1a58d7a>:0
      at Toggl.Core.UI.Navigation.NavigationService.Navigate[TViewModel,TInput,TOutput] (TInput payload, Toggl.Core.UI.Views.IView sourceView) [0x00273] in <d2bbbfc73143447c8d336e7fe1a58d7a>:0
      at Toggl.Core.UI.ViewModels.OnboardingViewModel.confirmCountryAndTermsOfService () [0x00068] in <d2bbbfc73143447c8d336e7fe1a58d7a>:0
      at Toggl.Core.UI.ViewModels.OnboardingViewModel.onGoogleLoginFailure (System.Exception exception) [0x000ba] in <d2bbbfc73143447c8d336e7fe1a58d7a>:0
      at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__7_0 (System.Object state) [0x00000] in <b7bb5f3538424ec1927304809f832185>:0
      at Android.App.SyncContext+<>c__DisplayClass2_0.<Post>b__0 () [0x00000] in <19bd7f1a5f294f918c3c66236f3a5fda>:0
      at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <19bd7f1a5f294f918c3c66236f3a5fda>:0
      at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <19bd7f1a5f294f918c3c66236f3a5fda>:0
        at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.9(intptr,intptr)
        at mono.java.lang.RunnableImplementor.n_run(Native Method)
        at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7682)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
```
Is this expected?

**Comment by daividssilverio** [comment]
@pdrj The actual wiring still needs to be done.
The way to test it right now is to navigate to the view model manually.
The crash happened because `TermsAndCountryViewModel` doesn't have the presenter setup yet. It's being done at #7138

File: Toggl.Droid/Activities/OnboardingActivity.ViewBinds.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            continueWithGoogleButton = FindViewById(Resource.Id.continueWithGoogleButton);
+            continueWithEmailButton = FindViewById<Button>(Resource.Id.continueWithEmailButton);
+            onboardingMessage = FindViewById<TextView>(Resource.Id.message);
+            loginGoogleLoginLabel = FindViewById<TextView>(Resource.Id.LoginGoogleLoginLabel);
+
+            onboardingMessage.Text = Shared.Resources.OnboardingMessageVariantA;
+            continueWithEmailButton.Text = Shared.Resources.ContinueWithEmail;
+            loginGoogleLoginLabel.Text = Shared.Resources.ContinueWithGoogle;
+            
+            onboardingSurfaceView.Holder.AddCallback(this);
```
Comment (pdrj) [inline Toggl.Droid/Activities/OnboardingActivity.ViewBinds.cs]: Are you sure it's not needed to do `RemoveCallback`?

File: Toggl.Droid/Resources/layout/OnboardingActivity.axml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            tools:text="Sign in With Google"
+            android:textColor="@color/onboardingContainedButtonTextColor"
+            android:textSize="17sp"
+            android:textStyle="normal" />
+    </LinearLayout>
+
+    <Button
+        android:id="@+id/continueWithEmailButton"
+        android:layout_width="match_parent"
+        android:layout_height="44dp"
```
Comment (pdrj) [inline Toggl.Droid/Resources/layout/OnboardingActivity.axml]: We're using lots of hardcoded heights here. Have we validated that this fits vertically on a small device?

**Comment by pdrj** [comment]
Sorry @daividssilverio, I went commit-by-commit and some of my comments are already not applicable ;). So please just don't care about these.

File: Toggl.Droid/Resources/drawable/outline_button.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<ripple xmlns:android="http://schemas.android.com/apk/res/android"
+    android:color="#33000000">
+    <item android:id="@android:id/mask">
+        <color android:color="@android:color/white" />
+    </item>
+    <item>
+        <shape android:shape="rectangle">
+            <corners android:radius="6px" />
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Droid/Resources/drawable/outline_button.xml]: Oh, definitely a mistake.

File: Toggl.Droid/Resources/values/Colors.xml [hunk 4]
```diff
@@ -107,7 +107,13 @@
   <color name="loginSignUpOptionsDivider">#F3F3F3</color>
 
   <!-- Onboarding -->
-  <color name="onboardingPagerBackgroundColor">#82D0CB</color>
+  <color name="onboardingBackgroundColor">#91DCD0</color>
+  <color name="onboardingOverlayBackgroundColor">#1A9DDDD3</color>
+  <color name="onboardingTogglManBorderHackColor">#CA91DCD0</color>
   <color name="onboardingOutlineButtonTextColor">#ffffff</color>
+  <color name="onboardingOutlineButtonStrokeColor">#ffffff</color>
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Droid/Resources/values/Colors.xml]: Are talking about the `camelCasing` or the `#ffffff`?

File: Toggl.Droid/Resources/layout/OnboardingActivity.axml (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/layout/OnboardingActivity.axml]: will test on reasonably tiny devices.

File: Toggl.Droid/Resources/layout/OnboardingActivity.axml [hunk 5]
```diff
@@ -4,17 +4,62 @@
...
     android:layout_height="match_parent"
-    android:background="@color/onboardingPagerBackgroundColor">
+    android:background="@color/onboardingBackgroundColor">
 
-    <androidx.viewpager2.widget.ViewPager2
-        android:id="@+id/onboardingPager"
-        android:layout_width="match_parent"
+    <SurfaceView
+        android:id="@+id/togglMan"
+        android:layout_width="275dp"
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Droid/Resources/layout/OnboardingActivity.axml]: Screenshot_1582912040

That's how it looks on a  320 x 480 mdpi device. Looks fine 🤔

File: Toggl.Droid/Resources/layout/OnboardingActivity.axml (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/layout/OnboardingActivity.axml]: It looks alright on smaller devices. Also, we need to give at least 48x48 touchable area (that's even a bit small).

File: Toggl.Droid/Activities/OnboardingActivity.ViewBinds.cs (hunk 1 above)
Comment (daividssilverio) [reply Toggl.Droid/Activities/OnboardingActivity.ViewBinds.cs]: Removed it `onDestroy`

File: Toggl.Droid/Resources/values/Colors.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (pdrj) [reply Toggl.Droid/Resources/values/Colors.xml]: The `#ffffff` - in the rest of the file, we use all caps for these definitions.

[the remark this answers is not in the document]
Comment (pdrj) [reply Toggl.Droid/Resources/values/Colors.xml]: But I see you already changed that ;).

File: Toggl.Droid/Resources/layout/OnboardingActivity.axml (hunk 5 above)
[the remark this answers is not in the document]
Comment (pdrj) [reply Toggl.Droid/Resources/layout/OnboardingActivity.axml]: Thanks for checking that 🙇

**Pull request merged**